src/main/java/duke/database/Database.java

```diff
@@ -0,0 +1,188 @@
+import duke.exception.DatabaseFileException;
+import duke.task.DeadlineTask;
+import duke.task.EventTask;
+import duke.task.Task;
+import duke.task.TaskType;
+import duke.task.ToDoTask;
+
+/**
+ * Encapsulates a database access and necessary methods for Duke's list.
+ */
+public abstract class Database {
+    /** Name of database table. */
+    protected static final String DATABASE_NAME = "tasks_data";
+    protected static final String TASK_TABLE_NAME = "tasks";
+    protected Connection connection;
+
+    /**
+     * Returns connection established by the database loader.
+     *
+     * @return SQL connection
+     */
+    public abstract Connection getSqlConnection();
+
+    /**
+     * Executes create table statement.
+     */
+    public abstract void load();
+
+    /**
+     * Initializes and tests SQL connection by attempting to execute select
+     * statements from respective tables in the database.
+     */
+    protected void initialize() {
+        connection = this.getSqlConnection();
+        try {
+            PreparedStatement ps = connection.prepareStatement("SELECT * FROM " + TASK_TABLE_NAME);
+            ResultSet rs = ps.executeQuery();
+            close(ps, rs);
+        } catch (SQLException ex) {
+            throw new DatabaseAccessException("Unable to access SQLite database...");
+        }
+    }
+
+    /**
+     * Returns the list of tasks stored in the database
+     *
+     * @return list of tasks stored
+     */
+    public abstract List<Task> getTasksList();
+
+    /**
+     * Adds a task to the database.
+     *
+     * @param task to be added
+     */
+    public abstract void addTask(Task task);
+
+    /**
+     * Removes a task from the database.
+     *
+     * @param index index of the task, {@code 1} for the first task, {@code 2} for
+     *              the second task etc.
+     * @return {@code null} if there are no tasks at the specified index
+     */
+    public abstract Task removeTask(int index);
+
+    /**
+     * Marks a task as completed in the database.
+     *
+     * @param index index of the task, {@code 1} for the first task, {@code 2} for
+     *              the second task etc.
+     * @return task that is marked completed, {@code null} if there are no tasks at
+     *         the specified index
+     */
+    public abstract Task markCompleted(int index);
+
+    /**
+     * Finds tasks by searching for a keyword/phrase.
+     *
+     * @param pattern keyword/phrase to be searched
+     * @return a list of tasks satisfying the condition mentioned above
+     */
+    public abstract List<Task> findTasksByName(String pattern);
+
+    /**
+     * Recreates a task based on provided information. Used when retrieving task
+     * from SQL database.
+     *
+     * @param type        type of task
+     * @param name        name of task
+     * @param isCompleted {@code true} if the task has been completed
+     * @param date        date of the task, nullable
+     * @return
```

Comment:
I think you might have forgotten to finish writing the comment for @return

Reply:
Haha, indeed...